pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1015,4 +1016,23 @@ public ConsumerStats getStats() {
 
     private static final Logger log = LoggerFactory.getLogger(ConsumerImpl.class);
 
+    @Override
+    public CompletableFuture<BrokerConsumerStats> getBrokerConsumerStatsAsync() {
+        if (getState() != State.Ready || !isConnected()) {
+            return FutureUtil.failedFuture(new PulsarClientException.NotConnectedException());
```

Comment:
PulsarClientException.NotConnectedException(ERROR_MSG) no such constructor exists 
I can add this constructor but I don't think the ERROR_MSG will be much different from the default message.